### PR TITLE
[DEV-11239] Adds new optional `formats` param to `GetStoryPageServerSideProps`

### PR DESCRIPTION
**Pull request by fgyimah**

### CODE REVIEW

File: packages/nextjs/src/adapter-nextjs/page-props/story.ts [hunk 1]
```diff
@@ -16,14 +17,15 @@ import type { PropsFunction } from './lib/types';
 
 export function getStoryPageServerSideProps<CustomProps extends Record<string, any>>(
     customProps: CustomProps | PropsFunction<CustomProps>,
+    formats?: Story.FormatVersion[],
```
Comment (e1himself) [inline packages/nextjs/src/adapter-nextjs/page-props/story.ts:20]: Do we need to do the same for `getStoryPageStaticProps()` below and `getStoryPreviewPageServerSideProps()` in the neighbouring storyPreview.ts file?

Comment (fgyimah) [reply packages/nextjs/src/adapter-nextjs/page-props/story.ts:20]: Yeah agree, making the update

Comment (fgyimah) [reply packages/nextjs/src/adapter-nextjs/page-props/story.ts:20]: Updated

**Review by e1himself — APPROVED** [review]
Looks good :+1:

**Pull request merged**